Seeders revised

Thanerik opened the pull request:
As the factories now is clearer, we revised our seeders to suit our current needs.
I brought the WignFaker provider back, to stimulating the words as well. Otherwise Faker would run out of uniqueness of the words.

**Aha moments:**
- If you in Factories NOT define `->create()->id`, then it doesn't create the model before overwritten in seeders. It's a good best practice to do it. If you chose to write `->create()->id` then it creates the models, even if you is overwritten it with another model later on

- You CANNOT create relation table (many-to-many) with only created_at, and then define the relations with `->withTimestamps()`. Even if it's `->withTimestamps(true, false)`. So I changed the behaviour to in some tables simply delete the timestamps (it's not necessary) - and in other I create the created_at column with default value of `now` when created, and let it be it. Now Laravel doesn't touch the timestamp, but it's inserted automatically in database. One disadvantage of that is the created_at column now is NOT in Carbon time format, but normal php DateTime format. We just need to bear that in mind.

Thanerik committed:
*Seeders revised*
As the factories now is clearer, we revised our seeders to suit our current needs.
I brought the WignFaker provider back, to stimulating the words as well. Otherwise Faker would run out of uniqueness of the words.

*Aha moments:*
- If you in Factories NOT define `->create()->id, then it doesn't create the model before overwritten in seeders. It's a good best practice to do it. If you chose to write `->create()->id` then it creates the models, even if you is overwritten it with another model later on

- You CANNOT create relation table (many-to-many) with only created_at, and then define the relations with `->withTimestamps()`. Even if it's `->withTimestamps(true, false)`. So I changed the behaviour to in some tables simply delete the timestamps (it's not necessary) - and in other I create the created_at column with default value of `now` when created, and let it be it. Now Laravel doesn't touch the timestamp, but it's inserted automatically in database. One disadvantage of that is the created_at column now is NOT in Carbon time format, but normal php DateTime format. We just need to bear that in mind.

Thanerik committed:
Apply fixes from StyleCI

Thanerik committed:
One line cleared in StopWatch

Thanerik commented:
When it's completed - create pull reqeust on the [lighthouse branch](https://github.com/Wign/Wign-design-v2/tree/feature/lighthouse)

Trudellik commented on app/Providers/FakerProvider.php:
```diff
@@ -0,0 +1,137 @@
+     * @return string the sentence
+     * @throws \Exception
+     */
+    public function wignWords(int $min = null, int $max = null)
```
Er det literal eller ren faktisk word som kører her?

Thanerik replied:
Det er ligegyldige. Her er det bare en provider til vores faker funktion, som sørger for at fylde den med en masse ord der kommer fra den oprindelig Wign. Det er fra denne funktion vi får de mange mærkelig ord-sammensætninger. Dens eneste og fornemmeste formål er at skabe dem, og det bruges kun i factories.

Trudellik commented on app/Translation.php:
```diff
@@ -62,6 +62,11 @@ public function editor()
         return $this->belongsTo('App\User', 'editor_id');
     }
 
+    public function likes()
```
Likes skal høre under sign. Likes vil netop blive nulstillet hver gang translation bliver opdateret med ny word, desc eller sign, hvilket er uhensigtmæssig. Sign har den tungeste betydningsvægt, hvorfor det vil give bedst mening med likes skal høre under sign.

Thanerik replied:
Ændret

Trudellik commented on database/factories/UserFactory.php:
```diff
@@ -1,43 +1,29 @@
-        'role_id' => $role,
+        'name'       => $faker->name,
+        'email'      => $faker->unique()->safeEmail,
+        'password'   => Hash::make($faker->password),
```
hvordan kan vi logge ind i denne bruger, hvis det er noget vi vil se fra denne brugers vinkel? Vi implementerer slet ikke password-beskyttelse selv, så det er ikke relevant os at tjekke hvorvidt login virker som vi ønsker.

Thanerik replied:
1. I UserTableSeeder har jeg allerede lavet en Admin bruger, og en User bruger - begge med "pass" som brugernavn. Disse kan vi bruge som vores default login til alle funktionalitet i systemet.
2. Ønsker du at tjekke én specifikt bruger i systemet og se hvordan denne agerer, så kan vi nemt tilføje denne pakke: https://packagist.org/packages/kabbouchi/nova-impersonate - der tillader en admin at logge ind som brugeren vha. Nova og kigge rundt som denne bruger

Trudellik commented on database/seeds/RequestsTableSeeder.php:
```diff
@@ -12,19 +13,10 @@ class RequestsTableSeeder extends Seeder
     public function run()
     {
-        $UsersTotal = User::count();
+        $users = User::all()->random(40);
```
antal tilknytning til request bør være varierende.

Thanerik replied:
Det sker ikke her. Her tager jeg bare 40 tilfældige brugere, og lade dem requeste én ord hver.

Thanerik replied:
Vil vi gerne have nogen der requester flere ord på én gang, laver vi bare en ny factory lige bagefter

Thanerik committed:
Merge branches 'develop' and 'feature/SeederRevised' of https://github.com/Wign/Wign-design-v2 into feature/SeederRevised

 Conflicts:
	_ide_helper.php
	app/Description.php
	app/Sign.php
	app/Tag.php
	app/Translation.php
	app/User.php
	app/Word.php

Thanerik committed:
Revised the models, seeders, factories to:
- Use a common Language model & table for both written and signed language
- Switch the likes to liking the signs directly: to preserve the mind model of liking a sign, and not liking a whole entity (translation), and making it easier to update translation model without breaking the likings

Thanerik committed:
Updated model phpDoc

Thanerik committed:
Adding ide_helper to ignore list. It should be generated locally using `composer install`

Thanerik committed:
Apply fixes from StyleCI

Trudellik reviewed (commented):
Nu er denne PR ultra lang fordi der er proppet ind med en masse ting. TIl næste vil det være rart at dele PR i sm å bidder, så dem som ikke kræver review, kan vi bare godkende selv.

Thanerik commented:
@Kennik89 jamen det VAR en lille pull request, indtil jeg måtte merge en masse ting ind pga. en opdateret develop-branch #13 (din 😉)
